libcontainer/utils/utils.go

```diff
@@ -50,19 +50,19 @@ func CleanPath(path string) string {
 
 	// Ensure that all paths are cleaned (especially problematic ones like
 	// "/../../../../../" which can cause lots of issues).
-	path = filepath.Clean(path)
+
+	if filepath.IsAbs(path) {
+		return filepath.Clean(path)
+	}
 
 	// If the path isn't absolute, we need to do more processing to fix paths
 	// such as "../../../../<etc>/some/path". We also shouldn't convert absolute
 	// paths to relative ones.
-	if !filepath.IsAbs(path) {
-		path = filepath.Clean(string(os.PathSeparator) + path)
-		// This can't fail, as (by definition) all paths are relative to root.
-		path, _ = filepath.Rel(string(os.PathSeparator), path)
-	}
+	path = filepath.Clean(string(os.PathSeparator) + path)
+	// This can't fail, as (by definition) all paths are relative to root.
+	path, _ = filepath.Rel(string(os.PathSeparator), path)
```

Comment:
What case are we handling with the previous `filepath.Clean()` that wouldn't be handled by just doing the filepath.Rel() call? I mean, what would not give the expected result if we remove the filepath.Clean call?

Reply:
The first call is dealing with leading `../../../../../` (it's not just `filepath.Clean`, it's `filepath.Clean("/"+path)`) -- `Rel` won't help with that.

Reply:
Ohh, makes sense, sorry!